app/controllers/businessmatching/updateservice/remove/RemoveBusinessTypesSummaryController.scala

```diff
@@ -16,23 +16,53 @@
 
 package controllers.businessmatching.updateservice.remove
 
+import cats.data.OptionT
+import cats.implicits._
 import connectors.DataCacheConnector
 import controllers.BaseController
+import controllers.businessmatching.updateservice.RemoveBusinessTypeHelper
+import forms.EmptyForm
 import javax.inject.Inject
+import models.flowmanagement.{RemoveBusinessTypeFlowModel, RemoveBusinessTypesSummaryPageId}
+import services.flowmanagement.Router
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.frontend.auth.AuthContext
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
+import views.html.businessmatching.updateservice.remove.remove_activities_summary
+
+import scala.concurrent.Future
 
 class RemoveBusinessTypesSummaryController @Inject()(
-                                                 val authConnector: AuthConnector,
-                                                 val dataCacheConnector: DataCacheConnector
-                                               ) extends BaseController {
+                                                   val authConnector: AuthConnector,
+                                                   val dataCacheConnector: DataCacheConnector,
+                                                   val helper: RemoveBusinessTypeHelper,
+                                                   val router: Router[RemoveBusinessTypeFlowModel]
+                                                 ) extends BaseController {
 
-  def get = Authorised.async{
+  def get = Authorised.async {
     implicit authContext =>
-      implicit request => ???
+      implicit request => {
+        for {
+          flow <- OptionT(dataCacheConnector.fetch[RemoveBusinessTypeFlowModel](RemoveBusinessTypeFlowModel.key))
+        } yield Ok(remove_activities_summary(EmptyForm, flow))
+      } getOrElse InternalServerError("Unable to get the flow model")
   }
 
-  def post = Authorised.async{
+  def post = Authorised.async {
     implicit authContext =>
-      implicit request => ???
+      implicit request => {
+        for {
+          model <- updateSubscription
+          route <- OptionT.liftF(router.getRoute(RemoveBusinessTypesSummaryPageId, model))
+        } yield route
+      } getOrElse InternalServerError("Unable to remove the business type")
   }
+
+  private def updateSubscription()
+                                (implicit ac: AuthContext, hc: HeaderCarrier): OptionT[Future, RemoveBusinessTypeFlowModel] = for {
+    model <- OptionT(dataCacheConnector.fetch[RemoveBusinessTypeFlowModel](RemoveBusinessTypeFlowModel.key))
+    _ <- helper.removeFitAndProper(model)
```

Comment:
Where does the different types of section data get removed?

Reply:
AMLS-3814 has been raised to deal with this issue.